examples/audio/uk_ireland_accent_recognition.py

```diff
@@ -0,0 +1,664 @@
+"""
+Title: UK & Ireland Accent Recognition Using Transfer Learning
+Author: [Fadi Badine](https://twitter.com/fadibadine)
+Date created: 16/04/2022
+Last modified: 16/04/2022
+Description: This notebook trains a model to classify UK & Ireland accents using feature extraction from Yamnet.
+"""
+
+"""
+# Prerequisits
+
+We need to install TensorFlow IO that we use to resample audio files to 16 kHz as
+required by Yamnet model
+"""
+
+"""shell
+pip install -U -q tensorflow_io
+"""
+
+"""
+# Configuration
+"""
+
+SEED = 1337
+EPOCHS = 100
+BATCH_SIZE = 64
+VALIDATION_RATIO = 0.1
+MODEL_NAME = "uk_irish_accent_recognition"
+
+# Location where the dataset will be downloaded.
+# By default (None), keras.utils.get_file will use /root/.keras/ as the CACHE_DIR
+CACHE_DIR = None
+
+# The location of the dataset
+URL_PATH = "https://www.openslr.org/resources/83/"
+
+# List of datasets compressed files that contain the audio files
+zip_files = {
+    0: "irish_english_male.zip",
+    1: "midlands_english_female.zip",
+    2: "midlands_english_male.zip",
+    3: "northern_english_female.zip",
+    4: "northern_english_male.zip",
+    5: "scottish_english_female.zip",
+    6: "scottish_english_male.zip",
+    7: "southern_english_female.zip",
+    8: "southern_english_male.zip",
+    9: "welsh_english_female.zip",
+    10: "welsh_english_male.zip",
+}
+
+# We see that there are 2 compressed files for each accent (except Irish):
+# - One for male speakers
+# - One for female speakers
+# However, we will using a gender agnostic dataset.
+
+# List of gender agnostic categories
+gender_agnostic_categories = [
+    "ir",  # Irish
+    "mi",  # Midlands
+    "no",  # Northern
+    "sc",  # Scottish
+    "so",  # Southern
+    "we",  # Welsh
+]
+
+class_names = [
+    "Irish",
+    "Midlands",
+    "Northern",
+    "Scottish",
+    "Southern",
+    "Welsh",
+    "Not a speech",
+]
+
+import os
+import io
+import csv
+import random
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+import tensorflow_hub as hub
+import tensorflow_io as tfio
+
+from tensorflow import keras
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from scipy import stats
+from IPython.display import Audio
+
+print(f"TensorFlow: {tf.__version__}")
+print(f"TensorFlow Hub: {hub.__version__}")
+print(f"TensorFlow IO: {tfio.__version__}")
```

Comment:
No need for these statements; if the code requires specific versions you can just list them in the introduction